sds_data_manager/lambda_code/SDSCode/indexer.py

```diff
@@ -20,6 +19,176 @@
+    with Session(engine) as session:
+        session.add(models.FileCatalog(**metadata_params))
+        # Event time looks like this:
+        # "time": "2024-01-16T17:35:08Z"
+
+        # Parses the time string from the event to a datetime object.
+        ingestion_date = datetime.datetime.strptime(event["time"], "%Y-%m-%dT%H:%M:%SZ")
+        # Formats the datetime object to a string with the format "%Y%m%d".
+        ingestion_data_str = ingestion_date.strftime("%Y%m%d")
+        # Parses the newly formatted string back into a datetime
+        # object with the desired structure.
+        ingestion_date_object = datetime.datetime.strptime(ingestion_data_str, "%Y%m%d")
+
+        status_params = {
+            "file_to_create_path": s3_filepath,
+            "status": models.Status.SUCCEEDED,
+            "job_definition": None,
+            "ingestion_date": ingestion_date_object,
+        }
+        session.add(models.StatusTracking(**status_params))
+        session.commit()
+    return http_response(status_code=200, body="Success")
+
+
+def batch_event_handler(event, engine):
+    """Handler for Batch event
+
+    Parameters
+    ----------
+    event : dict
+        The JSON formatted document with the data required for the
+        lambda function to process
+    engine : sqlalchemy.engine.Engine
+        Engine object to interact with the database.
+
+    Returns
+    -------
+    dict
+        HTTP response
+    """
+
+    if event["detail-type"] != "Batch Job State Change":
+        return http_response(status_code=400, body="Invalid event")
+    if event["detail"]["status"] == "SUCCEEDED":
+        # TODO:
+        # Get filename and then
+        # write it to file catalog table
+        pass
+
+    # TODO: update status table
+
+    return http_response(status_code=200, body="Success")
+
+
+def custom_event_handler(event, engine):
+    """_summary_
+
+    Parameters
+    ----------
+    event : dict
+        The JSON formatted document with the data required for the
+        lambda function to process
+    engine : sqlalchemy.engine.Engine
+        Engine object to interact with the database.
+
+    Returns
+    -------
+    dict
+        HTTP response
+    """
+    if event["detail-type"] != "Job Started":
+        msg = "Invalid event, must have detail-type 'Job Started'"
+        return http_response(status_code=400, body=msg)
+
+    if event["detail"]["status"] != "INPROGRESS":
+        return http_response(status_code=400, body="Invalid job status")
+
+    file_to_create = event["detail"]["file_to_create"]
+    filename = os.path.basename(file_to_create)
+    logger.info(f"Attempting to insert {filename} into database")
+
+    try:
+        _ = ScienceFile(filename)
+    except InvalidScienceFileError as e:
+        return http_response(status_code=400, body=str(e))
+
+    # Write event information to status tracking table.
+    logger.info(f"Inserting {filename} into database")
+    with Session(engine) as session:
```

Comment:
I'd like this call to be pulled out into a helper function in a different database focused file - so you can just call `database_handler.update_status_table(status_params)` instead of opening a database session here. (Gave similar advice to Laura)